impeller/renderer/backend/vulkan/context_vk.cc

```diff
@@ -484,6 +488,54 @@ ContextVK::ContextVK(
     return;
   }
 
+  //----------------------------------------------------------------------------
+  /// Setup the command pool.
+  ///
+  vk::CommandPoolCreateInfo graphics_command_pool_info;
+  graphics_command_pool_info.queueFamilyIndex = graphics_queue->index;
+  graphics_command_pool_info.flags = vk::CommandPoolCreateFlagBits::eTransient;
+  auto graphics_command_pool =
+      device.value->createCommandPoolUnique(graphics_command_pool_info);
+  if (graphics_command_pool.result != vk::Result::eSuccess) {
+    VALIDATION_LOG << "Could not create graphics command pool.";
+    return;
+  }
+
+  //----------------------------------------------------------------------------
+  /// Setup the descriptor pool. This needs to be dynamic but we just allocate a
+  /// jumbo pool and hope for the best.
+  ///
+  constexpr size_t kPoolSize = 1024 * 3;
```

Comment:
Should we have a bug tracking work to make this more robust? or is it unlikely to be a problem in practice.

Reply:
A pool maybe even a quarter this size ought to be more than sufficient for a typical Flutter application. But stress tests can blow past this budget and if we are going to be showing off Impeller use cases (like the Kaleidoscope thing) that specifically make a ton of draw calls, we can hit this hard cap. Terminating (not here but not being able to submit typically will make us tear down the whole thing) will probably be a bad look.

Filed https://github.com/flutter/flutter/issues/new